### ISSUE TITLE
Auto Lod algorithm causes seams between meshes

### ISSUE THREAD

**Issue by Arnklit** [godotengine/godot#59024]
### Godot version

4.0.alpha4

### System information

Windows 10, GLES3, GTX 980Ti

### Issue description

I've been looking at using the new auto lod system to set up terrain chunks, but the decimate algorithm used doesn't seem to favor keeping edges intact, so seams quickly become very visible. Would it be possible to add an option to favor conserving edges?

EDIT: Note I exaggerate the issue in this screenshot  by lowering the lod bias, but the issue is there with lod bias 1 as well.

![image](https://user-images.githubusercontent.com/4955051/157857862-cffc206b-c8c9-4c26-8c6a-720a4b21293a.png)

Let me know if this should be logged as a proposal instead, then I can close this and make a proposal instead.

### Steps to reproduce

Try out the attached example project

### Minimal reproduction project

[terrain_test2.zip](https://github.com/godotengine/godot/files/8231465/terrain_test2.zip)

**Comment by Calinou** [comment]
cc @fire

**Comment by fire** [comment]
I think upstream has a bug report too. https://github.com/zeux/meshoptimizer/issues/311

**Comment by Zylann** [comment]
The patch I applied for my case makes it so open edges are never simplified.
The way I done it is a "free" way (doesn't cost anything) but has the drawback of not having the option to switch to the original logic at runtime. I did it this way because I do this at runtime, and a lot.
I think if we want it to be an option, a bit more work needs to be done, in such a way zeux would consider it good enough for it to be part of the official library. I think it's not that hard to do, but maybe harder to decide what's best to preserve good performance (it's about changing a lookup table).
